core/src/main/java/org/apache/gravitino/authorization/AccessControlManager.java

```diff
@@ -143,42 +199,60 @@ public Role createRole(
       Map<String, String> properties,
       List<SecurableObject> securableObjects)
       throws RoleAlreadyExistsException, NoSuchMetalakeException {
-    return roleManager.createRole(metalake, role, properties, securableObjects);
+    return TreeLockUtils.doWithTreeLock(
+        NameIdentifier.of(AuthorizationUtils.ofRoleNamespace(metalake).levels()),
+        LockType.WRITE,
+        () -> roleManager.createRole(metalake, role, properties, securableObjects));
   }
 
   @Override
   public Role getRole(String metalake, String role)
       throws NoSuchRoleException, NoSuchMetalakeException {
-    return roleManager.getRole(metalake, role);
+    return TreeLockUtils.doWithTreeLock(
+        AuthorizationUtils.ofRole(metalake, role),
+        LockType.READ,
+        () -> roleManager.getRole(metalake, role));
   }
 
   @Override
   public boolean deleteRole(String metalake, String role) throws NoSuchMetalakeException {
-    return roleManager.deleteRole(metalake, role);
+    return TreeLockUtils.doWithTreeLock(
+        NameIdentifier.of(AuthorizationUtils.ofRoleNamespace(metalake).levels()),
+        LockType.WRITE,
+        () -> roleManager.deleteRole(metalake, role));
   }
 
   @Override
   public String[] listRoleNames(String metalake) throws NoSuchMetalakeException {
-    return roleManager.listRoleNames(metalake);
+    return TreeLockUtils.doWithTreeLock(
+        NameIdentifier.of(metalake), LockType.READ, () -> roleManager.listRoleNames(metalake));
```

Comment:
Should we use the role namespace like above?

Reply:
done